go/runtime/host/sgx/epid.go

```diff
@@ -79,6 +92,28 @@ func (ep *teeStateEPID) Update(ctx context.Context, sp *sgxProvisioner, conn pro
 		return nil, fmt.Errorf("error while verifying attestation evidence: %w", err)
 	}
 
+	// Decode the AVR so we can do further checks.
+	avr, decErr := cmnIAS.UnsafeDecodeAVR(avrBundle.Body)
+	if decErr == nil && avr.TCBEvaluationDataNumber < quotePolicy.MinTCBEvaluationDataNumber {
+		// Retry again with early updating.
+		evidence.EarlyTCBUpdate = true
+		avrBundle, err = sp.ias.VerifyEvidence(ctx, &evidence)
+		if err != nil {
+			return nil, fmt.Errorf("error while verifying attestation evidence with early update: %w", err)
+		}
+		avr, decErr = cmnIAS.UnsafeDecodeAVR(avrBundle.Body)
+	}
+	if decErr != nil {
+		return nil, fmt.Errorf("unable to decode AVR: %w", decErr)
+	}
+	if avr.TCBEvaluationDataNumber < quotePolicy.MinTCBEvaluationDataNumber {
+		return nil, fmt.Errorf(
+			"AVR TCB data evaluation number invalid (%v >= %v)",
+			avr.TCBEvaluationDataNumber,
+			quotePolicy.MinTCBEvaluationDataNumber,
```

Comment:
The positions of the two should ne switched or the error string should be changed.